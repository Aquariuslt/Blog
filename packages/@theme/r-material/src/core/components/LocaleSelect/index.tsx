import * as React from 'react';
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';

import Select from '@material-ui/core/Select';
import MenuItem from '@material-ui/core/MenuItem';
import InputLabel from '@material-ui/core/InputLabel';

import { useTranslation } from 'react-i18next';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      display: 'flex',
      flexWrap: 'wrap',
      margin: theme.spacing(1)
    },
    label: {
      margin: theme.spacing(0, 1)
    },
    select: {
      width: '100%',
      margin: theme.spacing(0, 1)
    }
  })
);

const LocaleSelect = () => {
  const classes = useStyles();
  const [t, i18n] = useTranslation();

  const changeLanguage = (lang) => {
    i18n.changeLanguage(lang).then();
  };

  return (
    <div className={classes.root}>
      <InputLabel className={classes.label} htmlFor="select-locale">
        {t('core.locale.select.label')}
      </InputLabel>
      <Select
        className={classes.select}
        inputProps={{
          name: 'locale',
          id: 'select-locale'
        }}
        aria-label={t('core.locale.select.label')}
        value={i18n.language}
        onChange={(event) => changeLanguage(event.target.value)}
      >
        <MenuItem value="zh">{t('core.locale.select.option.zh')}</MenuItem>
        <MenuItem value="en">{t('core.locale.select.option.en')}</MenuItem>
      </Select>
    </div>
  );
};

export default LocaleSelect;
