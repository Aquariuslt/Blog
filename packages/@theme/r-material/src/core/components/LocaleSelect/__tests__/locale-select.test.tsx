import * as React from 'react';

import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';

import LocaleSelect from '../';

Enzyme.configure({ adapter: new Adapter() });

describe('@theme/r-material: core/components: LocaleSelect', () => {
  it('# should mount LocaleSelect component w/o i18next options correctly', () => {
    const wrapper = Enzyme.shallow(<LocaleSelect />);
    expect(wrapper).toMatchSnapshot();
  });

  it('# should change locale when click and select language select menu', async () => {
    await import('./i18n-for-tests');

    const wrapper = Enzyme.shallow(<LocaleSelect />);
    wrapper.find('[aria-label="Language"]').simulate('change', {
      target: {
        value: 'zh'
      }
    });
    expect(wrapper).toMatchSnapshot();
  });
});
