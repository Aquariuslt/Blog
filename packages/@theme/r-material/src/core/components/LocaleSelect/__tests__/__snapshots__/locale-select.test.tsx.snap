// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@theme/r-material: core/components: LocaleSelect # should change locale when click and select language select menu 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(InputLabel))
    className="makeStyles-label-2"
    htmlFor="select-locale"
  >
    Language
  </WithStyles(ForwardRef(InputLabel))>
  <WithStyles(ForwardRef(Select))
    aria-label="Language"
    className="makeStyles-select-3"
    inputProps={
      Object {
        "id": "select-locale",
        "name": "locale",
      }
    }
    onChange={[Function]}
    value="en"
  >
    <WithStyles(ForwardRef(MenuItem))
      value="zh"
    >
      Chinese
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      value="en"
    >
      English
    </WithStyles(ForwardRef(MenuItem))>
  </WithStyles(ForwardRef(Select))>
</div>
`;

exports[`@theme/r-material: core/components: LocaleSelect # should mount LocaleSelect component w/o i18next options correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(InputLabel))
    className="makeStyles-label-2"
    htmlFor="select-locale"
  >
    core.locale.select.label
  </WithStyles(ForwardRef(InputLabel))>
  <WithStyles(ForwardRef(Select))
    aria-label="core.locale.select.label"
    className="makeStyles-select-3"
    inputProps={
      Object {
        "id": "select-locale",
        "name": "locale",
      }
    }
    onChange={[Function]}
  >
    <WithStyles(ForwardRef(MenuItem))
      value="zh"
    >
      core.locale.select.option.zh
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))
      value="en"
    >
      core.locale.select.option.en
    </WithStyles(ForwardRef(MenuItem))>
  </WithStyles(ForwardRef(Select))>
</div>
`;
